beam-envi-reader/src/main/java/org/esa/beam/dataio/envi/EnviProductWriter.java

```diff
@@ -104,7 +104,7 @@ public void writeBandRasterData(Band sourceBand,
         checkSourceRegionInsideBandRegion(sourceWidth, sourceBandWidth, sourceHeight, sourceBandHeight, sourceOffsetX,
                                           sourceOffsetY);
         final ImageOutputStream outputStream = getOrCreateImageOutputStream(sourceBand);
-        long outputPos = sourceOffsetY * sourceBandWidth + sourceOffsetX;
+        long outputPos = (long) sourceOffsetY * (long) sourceBandWidth + sourceOffsetX;
```

Comment:
Yes, it's correct: that change was already included in master and snap-1.1.x branches, but not in snap-1.0.x